### PR TITLE
DS3231 v2

### PR DESCRIPTION
**Pull request by bhuvanchandra**
Changes since #166  :

- Use enums instead of defines.
- Add license string in the ds3231 driver.

### CODE REVIEW

File: extras/ds3231/component.mk [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Component makefile for extras/ds3231
+
+# expected anyone using bmp driver includes it as 'ds3231/ds3231.h'
```
Comment (sheinz) [inline extras/ds3231/component.mk]: **bmp** driver

**Comment by sheinz** [comment]
I'm not sure if we follow any coding style here in EOR,  but using tabs for indentation is not recommended, at least for C programming.

**Comment by UncleRus** [comment]
Coding style requirements are described in readme:
https://github.com/SuperHouse/esp-open-rtos#coding-style

File: extras/ds3231/component.mk (hunk 1 above)
Comment (bhuvanchandra) [inline extras/ds3231/component.mk]: I took the bmp driver dir as skeleton, missed updating the comment. Will fix this in next version.

**Pull request closed without merging**